pkg/controller/clusterinstallation/mysql.go

```diff
@@ -76,12 +76,30 @@ func (r *ReconcileClusterInstallation) createMySQLClusterIfNotExists(mattermost
 	return nil
 }
 
-// getOrCreateMySQLSecrets get or create the MySQL secrets used by MySQL Operator to spin the cluster and
-// also used by mattermost to get the credentials to use to access the DB.
-// dbData is a []string -> { DBUserName, DBUserPassword, DBName }
 func (r *ReconcileClusterInstallation) getOrCreateMySQLSecrets(mattermost *mattermostv1alpha1.ClusterInstallation, reqLogger logr.Logger) (databaseInfo, error) {
-	dbSecretName := fmt.Sprintf("%s-mysql-root-password", mattermost.Name)
 	dbSecret := &corev1.Secret{}
+	dbInfo := databaseInfo{
+		userName:     "",
+		userPassword: "",
+		dbName:       "",
+	}
+	// Check if the custom(existing) MySQL secret is provided
+	if mattermost.Spec.Database.ExistingSecret != "" {
+		if err := r.client.Get(context.TODO(), types.NamespacedName{
+			Namespace: mattermost.Namespace,
+			Name:      mattermost.Spec.Database.ExistingSecret,
+		}, dbSecret); err != nil {
+			return dbInfo, errors.Wrap(err, "unable to locate custom/existing MySQL secret")
+		}
+		dbInfo = databaseInfo{
+			userName:     string(dbSecret.Data["USER"]),
```

Comment:
I think we should check that these three secret values are not empty before returning. If any of them are empty, we should return an error.

Reply:
I made use of the existing `dbInfo.Valid()` which will return an error if any of the secret values are empty. 
https://github.com/mattermost/mattermost-operator/blob/c85abeaa8d19960b2eb9fbd66e7107b8cd9a3551/pkg/controller/clusterinstallation/mysql.go#L94-L99

Reply:
Oh, missed that. Looks good!